### PR TITLE
Fix updating the Git working tree for outdated branches

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Please see the commit messages for details.

### CODE REVIEW

File: downloader/src/funTest/kotlin/vcs/GitWorkingTreeFunTest.kt [hunk 1]
```diff
@@ -21,62 +21,62 @@ package org.ossreviewtoolkit.downloader.vcs
...
+        revision = "main"
+    )
+
+    lateinit var workingTree: WorkingTree
 
     beforeSpec {
-        val zipFile = getAssetFile("pipdeptree-2018-01-03-git.zip")
-        println("Extracting '$zipFile' to '$zipContentDir'...")
-        zipFile.unpack(zipContentDir)
+        println("Cloning '${vcsInfo.url}' to '$repoDir'...")
```
Comment (sschuberth) [inline downloader/src/funTest/kotlin/vcs/GitWorkingTreeFunTest.kt:48]: Nit: We usually don't quite URLs in console output.

File: downloader/src/main/kotlin/vcs/Git.kt [hunk 2]
```diff
@@ -234,7 +237,13 @@ class Git : VersionControlSystem(), CommandLineTool {
             logger.warn { "Failed to fetch everything: ${it.collectMessages()}" }
         }.mapCatching {
             // TODO: Migrate this to JGit once https://bugs.eclipse.org/bugs/show_bug.cgi?id=383772 is implemented.
-            run("checkout", revision, workingDir = git.repository.workTree)
+            run("checkout", revision, workingDir = workingTree.workingDir)
```
Comment (sschuberth) [inline downloader/src/main/kotlin/vcs/Git.kt:240]: Why was the `workingDir` argument change necessary?

Comment (mnonnenmacher) [reply downloader/src/main/kotlin/vcs/Git.kt:240]: It's wasn't necessary, I changed it for consistency as the `workingTree` is now available as a function parameter.

Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/Git.kt:240]: Could you please briefly mention this in the commit message, to make clear that this part of the diff is not supposed to change behavior?

Comment (mnonnenmacher) [reply downloader/src/main/kotlin/vcs/Git.kt:240]: I have moved it to a separate commit.

Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/Git.kt:240]: Even better, thanks!

**Review by sschuberth — APPROVED** [review]

**Pull request merged**